tensorflow/core/kernels/dml_scatter_nd_op.cc

```diff
@@ -0,0 +1,309 @@
+      OP_REQUIRES(
+          ctx,
+          updates.shape().dim_size(i + outer_dims) == shape.dim_size(ix + i),
+          errors::InvalidArgument(
+              "The inner ", shape.dims() - ix,
+              " dimensions of output.shape=", shape.DebugString(),
+              " must match the inner ", updates.shape().dims() - outer_dims,
+              " dimensions of updates.shape=", updates.shape().DebugString()));
+    }
+    OP_REQUIRES(ctx, shape_input.dims() == 1,
+                errors::InvalidArgument("Shape must be a vector"));
+
+    OP_REQUIRES_OK(ctx, ValidateInputs<Index>(shape, indices, updates));
+  }
+};
+
+template <typename Index>
+class DmlScatterNdKernel : public DmlKernel {
+ public:
+  using InitHelper = ScatterNdInitHelper<Index>;
+
+  DmlScatterNdKernel(DmlKernelConstruction* ctx,
+                     const InitHelper* init_helper) {
+    const TensorShape& in_out_shape = ctx->GetOutputTensorShape(0);
+    const TensorShape& indices_shape = ctx->GetInputTensorShape(0);
+    const TensorShape& updates_shape = ctx->GetInputTensorShape(1);
+
+    DmlTensorInfo in_out_tensor;
+    in_out_tensor.desc = DmlTensorDesc::Create(ctx->GetOutputDataType(0),
+                                               in_out_shape, in_out_shape);
+
+    DmlTensorInfo indices_tensor;
+    indices_tensor.desc = DmlTensorDesc::Create(ctx->GetInputDataType(0),
+                                                indices_shape, indices_shape);
+
+    DmlTensorInfo updates_tensor;
+    updates_tensor.desc = DmlTensorDesc::Create(ctx->GetInputDataType(1),
+                                                updates_shape, updates_shape);
+
+    DmlKernelTensors tensors;
+    tensors.inputs = {in_out_tensor, indices_tensor, updates_tensor};
+    tensors.outputs = {in_out_tensor};
+
+    auto input_descs = GetDmlTensorDescs(tensors.inputs);
+    auto output_descs = GetDmlTensorDescs(tensors.outputs);
+
+    DML_SCATTER_ND_OPERATOR_DESC scatter_nd_desc = {};
+    scatter_nd_desc.InputTensor = &input_descs[0];
+    scatter_nd_desc.IndicesTensor = &input_descs[1];
+    scatter_nd_desc.UpdatesTensor = &input_descs[2];
+    scatter_nd_desc.OutputTensor = &output_descs[0];
+    scatter_nd_desc.InputDimensionCount = in_out_shape.dims();
+    scatter_nd_desc.IndicesDimensionCount = indices_shape.dims();
+
+    DML_OPERATOR_DESC op_desc = {DML_OPERATOR_SCATTER_ND, &scatter_nd_desc};
+    Initialize(ctx, std::move(tensors), op_desc);
+  }
+
+  StatusOr<DmlGpuEvent> Compute(DmlKernelContext* ctx) const override {
+    D3D12BufferRegion indices_buffer =
+        ctx->CreateBufferForTensor(ctx->GetInputTensor(0));
+
+    D3D12BufferRegion updates_buffer =
+        ctx->CreateBufferForTensor(ctx->GetInputTensor(1));
+
+    D3D12BufferRegion output_buffer =
+        ctx->CreateBufferForTensor(*ctx->GetOutputTensor(0));
+
+    DmlBuffer params_buffer =
+        ctx->AllocateDefaultBuffer(output_buffer.SizeInBytes());
```

Comment:
Since this is a tensor full of zero's, it'd be better if this was a broadcasted scalar.